Scripts/RemediationScripts/Remediate-SetServiceBusMinReqTLSVersion.ps1

```diff
@@ -0,0 +1,779 @@
+                }
+            }
+        }
+    }
+
+    $serviceBusNamespacesCount = ($serviceBusNamespacesDetails | Measure-Object).Count
+
+    if ($serviceBusNamespacesCount -eq 0) {
+        Write-Host "No Azure Service Bus Namespace(s) found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+
+    Write-Host "Found [$($serviceBusNamespacesCount)] Azure Service Bus Namespace(s)." -ForegroundColor $([Constants]::MessageType.Update)
+
+    Write-Host $([Constants]::SingleDashLine)
+
+    # List for actionable service bus namespace(s)
+    $actionableServiceBusNamespacesDetails = @()
+
+    Write-Host "Filtering Azure Service Bus Namespace(s) for which secure TLS version is not secure..." -ForegroundColor $([Constants]::MessageType.Info)
+
+    $serviceBusNamespacesDetails | ForEach-Object {
+        if (-not (CheckIfCurrentTLSVersionIsSecure($_.MinimumTLSVersion))) {
+            $actionableServiceBusNamespacesDetails += $_
+        }
+    }
+
+    $actionableServiceBusNamespacesDetailsCount = ($actionableServiceBusNamespacesDetails | Measure-Object).Count
+
+    if ($actionableServiceBusNamespacesDetailsCount -eq 0) {
+        Write-Host "No Azure Service Bus Namespaces(s) found with non-secure TLS version enabled.. Exiting..." -ForegroundColor $([Constants]::MessageType.Warning)
+        Write-Host $([Constants]::DoubleDashLine)	
+        return
+    }
+
+    Write-Host "Found [$($actionableServiceBusNamespacesDetailsCount )] Azure Service Bus Namespace(s) for which non secure TLS version is enabled ." -ForegroundColor $([Constants]::MessageType.Update)
+    Write-Host $([Constants]::SingleDashLine)	
+
+    $colsProperty = @{Expression = { $_.ResourceName }; Label = "ResourceName"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.ResourceGroupName }; Label = "ResourceGroupName"; Width = 30; Alignment = "left" },
+    @{Expression = { $_.ResourceId }; Label = "ResourceId"; Width = 100; Alignment = "left" },
+    @{Expression = { $_.MinimumTLSVersion }; Label = "MinimumTLSVersion"; Width = 100; Alignment = "left" }
+
+    if (-not $AutoRemediation) {
+        Write-Host "Azure Service Bus Namespace(s) with non-secure TLS version enabled are:"
+        $actionableServiceBusNamespacesDetails  | Format-Table -Property $colsProperty -Wrap
+        Write-Host $([Constants]::SingleDashLine)
+    }
+
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\Set-SecureTLSVersionForServiceBusNamespaces"
+
+    if (-not (Test-Path -Path $backupFolderPath)) {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+
+    Write-Host "[Step 3 of 4] Back up Azure Service Bus Namespace(s) details..."
```

Comment:
Since this is a step, no need to add '...' at the end.

Reply:
Corrected.